### DISCUSSION TITLE
How does Apollo's Sim Control run perception/planning module without reading sensor data?

### DISCUSSION THREAD

**Discussion by chiragr15** [ApolloAuto/apollo#15367] [Q&A]
I am able to run some simple routing demos using bootstrap (Sim Control) on some demo maps. How is this possible without me sending any bag file with sensor data?
Since the prediction and perception modules require sensor data which in-turn gets utilized for planning and control, how does the toolchain get this sensor data? Does Sim Control provide some pre-simulated or trained sensor data which is used every time?

**Comment by daohu527** [comment]
No, sim-control for now don't consider obstcales, it is just used to simulate the planning module and generate paths.

We are adding obstacles in the scene, allowing users to edit the scene, such as the obstacle's path and speed. But sim control is still limited to testing the planning module
